build/local_kubernetes.sh

```diff
@@ -73,14 +73,43 @@ install_csi_hostpath_driver() {
     kubectl apply -fhttps://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/${SNAPSHOTTER_VERSION}/deploy/kubernetes/snapshot-controller/{rbac-snapshot-controller.yaml,setup-snapshot-controller.yaml}
 
     # Deploy the CSI Hostpath Driver
-    cd /tmp
-    git clone https://github.com/kubernetes-csi/csi-driver-host-path.git
-    cd csi-driver-host-path
-    sed -i 's/mountPropagation: Bidirectional/\#mountPropagation: Bidirectional/g' deploy/kubernetes-latest/hostpath/csi-hostpath-plugin.yaml
-    ./deploy/kubernetes-latest/deploy.sh
-
-    # Create StorageClass
-    kubectl apply -f ./examples/csi-storageclass.yaml
+    pushd /tmp
+      git clone https://github.com/kubernetes-csi/csi-driver-host-path.git
+      pushd csi-driver-host-path
+        sed -i 's/mountPropagation: Bidirectional/\#mountPropagation: Bidirectional/g' deploy/kubernetes-latest/hostpath/csi-hostpath-plugin.yaml
+
+        ./deploy/kubernetes-latest/deploy.sh
+
+        # Create StorageClass
+        kubectl apply -f ./examples/csi-storageclass.yaml
+      popd
+    popd
+}
+
+check_csi_hostpath_driver_installed() {
+    # Check VolumeSnapshot CRDs are installed
+    if ! kubectl diff -fhttps://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/${SNAPSHOTTER_VERSION}/client/config/crd/snapshot.storage.k8s.io_{volumesnapshots.yaml,volumesnapshotclasses.yaml,volumesnapshotcontents.yaml} 2>&1 > /dev/null ; then
+        echo "VolumeSnapshot CRDs are not installed."
+        exit 1
+    fi
+
+    # Check snapshot controller created
+    if ! kubectl diff -fhttps://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/${SNAPSHOTTER_VERSION}/deploy/kubernetes/snapshot-controller/{rbac-snapshot-controller.yaml,setup-snapshot-controller.yaml} 2>&1 > /dev/null ; then
```

Comment:
how would `diff` work if the snapshot controller  is already installed using different `SNAPSHOT_VERSION` and we have updated `SNAPSHOT_VERSION` to newer versions?
have we considered that, would that work as expected?

Reply:
Well, it's good question. But now we don't check existent of snapshot controller at all, thus people most likely don't even know that it have to be installed before running a tests. If version of snapshot controller is set somewhere in YAML and already installed version differs from the actual version, the `diff` will treat it as difference, so this check gonna be fail.